### PR TITLE
bugfix: fix create volume failed but metadata is stored

### LINKED ISSUE #703: [bug] pouch volume create failed, but metadata is stored
### Ⅰ. Issue Description
Create a volume failed, but `pouch volume list` could get the volume name.

### Ⅱ. Describe what happened

```
#touch /tmp/test6

#pouch volume create -d local  -o mount=/tmp -n test6
Error: {"message":"mount path is not a dir /tmp/test6"}


#echo $?
1

#pouch volume list
Name:
b2f197657d8fb9e03b4db314654449fee45328992e179cff2b94bfdd6e8b67d4
cb4b391c5d1b560c1d613658c53d16d60fdfd74c1ec198d43612ec9ee16e61a5
pouch-volume
test
test2
test3
test4
test5
test6

```
### Ⅲ. Describe what you expected to happen
`pouch list` should not contains `test6`

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1.
2.
3.

### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by rudyfly**
### Ⅰ. Describe what this PR did
fix create volume failed, but metadata is stored.

### Ⅱ. Does this pull request fix one issue?

fixes #703 

### Ⅲ. Describe how you did it
storing metadata after creating volume successfully.

### Ⅳ. Describe how to verify it
```bash
# touch /tmp/test6
# pouch volume create -d local  -o mount=/tmp -n test6
Error: {"message":"mount path is not a dir /tmp/test6"}

# echo $?
1

# pouch volume list
Name:
pouch-volume
```

### Ⅴ. Special notes for reviews

Signed-off-by: Rudy Zhang <rudyflyzhang@gmail.com>

### CODE REVIEW

File: volume/core.go [hunk 1]
```diff
@@ -154,11 +150,24 @@ func (c *Core) CreateVolume(id types.VolumeID) error {
...
 		}
+
+		if err := store.MetaPut(v); err != nil {
+			return err
+		}
 	}
 
 	if f, ok := dv.(driver.Formator); ok {
-		return f.Format(ctx, v, s)
+		if err := f.Format(ctx, v, s); err != nil {
```
Comment (allencloud) [inline volume/core.go]: I suggest that we could use the rule `return fast` to reduce idents and increase readability. So the code style would be:
```
f, ok := dv.(driver.Formator); 
if !ok {
    return nil
}

err = f.Format(ctx, v, s)
if err == nil {
    return nil
}
logrus.Errorf("failed to format new volume: %s, err: %v", v.Name, err)
logrus.Warnf("rollback create volume, start to cleanup new volume: %s", v.Name)
if err := c.RemoveVolume(id); err != nil {
	logrus.Errorf("failed to rollback create volume, cleanup new volume: %s, err: %v", v.Name, err)
    return err
}
return nil
```

Comment (rudyfly) [reply volume/core.go]: ok

**Comment by rudyfly** [comment]
@allencloud PTAL

**Comment by allencloud** [comment]
I suggest that if we fix an bug, we could add a corresponding test case to cover this. Then We can make sure it will never happen again in the CI  system to guarantee the quality. @rudyfly 
We need to add the test in the following ones. @rudyfly 
LGTM

**Pull request merged**